tests/MailboxTest.php

```diff
@@ -108,4 +108,30 @@ public function testCustomStatusFlag()
         $this->assertSame(3, $status->messages);
         $this->assertFalse(isset($status->uidnext), 'uidnext shouldn\'t be set');
     }
+
+    public function testBulkSetFlags()
+    {
+        $messages = [1 => 'Message A', 2 => 'Message B', 3 => 'Message C'];
+
+        foreach ($messages as $uid => $subject) {
+            $this->createTestMessage($this->mailbox, $subject);
+
+            $message = $this->mailbox->getMessage($uid);
+            $this->assertFalse($message->isFlagged());
+        }
+
+        $this->mailbox->setFlag('\\Flagged', \array_keys($messages));
```

Comment:
I've tested that creating the messages in different mailboxes and using the setFlag method on a random mailbox, the flags are changed to all the UIDs specificed because, as the name said, the flags are changed by UIDs regardless on the mailbox the mails are in.

As such I find misleading to have these two new methods in the `Mailbox` class since they don't depend on it.

What about moving the code (and the test) to the `Connection`?

Reply:
Check 189815aaa862758ab0bb040a3e08437572a14d45, UIDs are related with specific mailbox. Moving functionality to `Connection` would require a `mailboxName` parameter.